packages/picasso.js/src/web/renderer/canvas-renderer/shapes/text.js

```diff
@@ -4,11 +4,7 @@ import { detectTextDirection, flipTextAnchor } from '../../../../core/utils/rtl-
 
 export default function render(t, { g, ellipsed, doStroke }) {
   const text = ellipsed || ellipsText(t, measureText);
-  if (t['font-weight']) {
-    g.font = `${t['font-weight']} ${t['font-size']} ${t['font-family']}`;
-  } else {
-    g.font = `${t['font-size']} ${t['font-family']}`;
-  }
+  g.font = [t['font-style'], t['font-weight'], t['font-size'], t['font-family']].filter((v) => !!v).join(' ');
```

Comment:
Nitpicking here but it does not seem optional from a performance perspective to
1. Create an array
2. `filter` to create another array
3. `join` array

...everytime we render a text.

I would think [string concat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/concat) is a faster and provides a somewhat clean syntax. Some simpified tested in the browser console at least gave to a 50% performance increase compered to current method.

Example:
```js
g.font = (t["font-style"] ?? "").concat(t['font-weight'] ?? ", t['font-size'] ?? "", t['font-family'] ?? "")
```